src/saltext/azurerm/utils/azurerm.py

```diff
@@ -99,14 +99,18 @@ def _determine_auth(**kwargs):
         )
 
     try:
-        credentials = DefaultAzureCredential(authority=authority, **kwargs)
+        if "client_id" in kwargs and "tenant" in kwargs and "secret" in kwargs:
+            credentials = get_identity_credentials(**kwargs)
+        else:
+            credentials = DefaultAzureCredential(authority=authority, **kwargs)
     except ClientAuthenticationError:
-        raise SaltInvocationError(  # pylint: disable=raise-missing-from
+        raise SaltInvocationError(
             "Unable to determine credentials. "
             "A subscription_id with username and password, "
             "or client_id, secret, and tenant or a profile with the "
             "required parameters populated"
-        )
+        ) from None
```

Comment:
**issue (blocking):** if the `raise` is just going to be from `None`, just put the pylint disable statement back and remove the `from None`